homeassistant/components/zwave/__init__.py

```diff
@@ -311,30 +321,20 @@ def value_added(node, value):
 
     def node_added(node):
         """Handle a new node on the network."""
-        entity = ZWaveNodeEntity(node, network)
-        node_config = device_config.get(entity.entity_id)
+        entity = ZWaveNodeEntity(node, network, new_entity_ids)
+        name = node_name(node)
+        if new_entity_ids:
+            generated_id = generate_entity_id(DOMAIN + '.{}', name, [])
+        else:
+            generated_id = entity.entity_id
+        node_config = device_config.get(generated_id)
```

Comment:
So for the first point, unfortunately the ID post collision detection can't be used to ignore a device, since that ID is only available after the entity is added to hass. Colliding entity_ids aren't really meant to be used long-term anyway, so after they are renamed it won't be an issue.

For the second point, what do you think about exposing the value ID in the device state attributes, then including it in the rename card drop-down?